### DISCUSSION TITLE
Allow specifying devices in Assist commands

### DISCUSSION THREAD

**Discussion by tetele** [home-assistant/architecture#1060] [Entity Models]
# End goal

I think that devices should be included in the selection criteria that the default (Home Assistant) conversation agent uses to target entities. For example, one should be able to ask `what is the temperature in the fridge?` (where `fridge` is the name of a device having a `device_class: temperature` sensor) or `turn up the volume for the Sonos in the living room` (where `Sonos` is a device in the `living room` area, having a `media_player` entity)

## Requirements

Mandatory requirements
- the `device_registry` needs to be searchable by name
- the `default_agent` needs to pass a list of devices to the intent recognition engine
- matching of entities needs to be able to account for parent devices

Optional requirements
- `DeviceEntry`s should get a boolean flag making them exposable or not
- `DeviceEntry`s should have aliases

## Proposals

- Implement search capability for the device registry https://github.com/home-assistant/core/pull/113076
- Implement aliases for devices
- Implement exposure toggles for devices (separated by "types" of assistants like for other types of data)
- Implement device marching capabilities in Hassil and the default conversation agent
- Add sentences targeting devices in the `intents` repo

**Comment by joostlek** [comment]
How do you tackle devices with more of the same device classes? For example a fridge that also measures the outside temperature is usually also made with the device class Temperature. The only difference would be that that one probably has a name set.

**Reply by tetele** [reply]
We have the same issue atm with multiple entities with the same name. The current solution is to get the first such entity, whatever that means.

As a side note, i would imagine that if you have a device with multiple `sensor`s of the same `device_class`, you would not ask for one. For example if i have a multi-room temperature controller with sensors in each room, i would not ask `what is the temperature in the controller`, as i know it reports more than one.
